test/integration/puppetlabs/services/legacy_routes/legacy_routes_test.clj

```diff
@@ -31,20 +31,22 @@
 
 (deftest ^:integration legacy-routes
   (testing "The legacy web routing service properly handles old routes."
-    (bootstrap/with-puppetserver-running app
-      {}
-      (is (= 200 (:status (http-get "/v2.0/environments"))))
-      (is (= 200 (:status (http-get "/production/node/localhost"))))
-      (is (= 200 (:status (http-get "/production/certificate_statuses/all")))))))
+    (bootstrap/with-puppetserver-running-with-mock-jrubies
+     app
+     {}
+     (is (= 200 (:status (http-get "/v2.0/environments"))))
+     (is (= 200 (:status (http-get "/production/node/localhost"))))
+     (is (= 200 (:status (http-get "/production/certificate_statuses/all")))))))
 
 (deftest ^:integration old-master-route-config
   (testing "The old map-style route configuration map still works."
-    (bootstrap/with-puppetserver-running app
-      {:web-router-service
-       {:puppetlabs.services.master.master-service/master-service
-        {:master-routes "/puppet"
-         :invalid-in-puppet-4 "/"}}}
-      (is (= 200 (:status (http-get "/puppet/v3/node/localhost?environment=production"))))))
+    (bootstrap/with-puppetserver-running-with-mock-jrubies
+     app
+     {:web-router-service
+      {:puppetlabs.services.master.master-service/master-service
+       {:master-routes "/puppet"
+        :invalid-in-puppet-4 "/"}}}
+     (is (= 200 (:status (http-get "/puppet/v3/node/localhost?environment=production"))))))
```

Comment:
This one I think I'd be OK with the mocking because there is nothing about the request handler that is relevant to this test, but it'd be better if we could just make a request that doesn't trigger the request handler, so that the mocked request handler could still throw an exception.  I don't recall if there are any routes under `/puppet` that are clojure-only though :/